lib/mini_mime.rb

```diff
@@ -50,6 +50,13 @@ def binary?
   end
 
   class Db
+    LOCK = Mutex.new
+    @db = nil
+
+    def self.db
+      @db || LOCK.synchronize { @db ||= new }
```

Comment:
```suggestion

    def self.db
      @db ||= LOCK.synchronize { @db ||= new }
```

If you want to avoid the warning on older rubies. But both works I guess.

Reply:
Yeah, I meant alternative solution to that same warning.

Reply:
I am ok with the implementation, its a bit of hair splitting going one way vs the other.